src/Data/notifyOnConnectionTimeout.tsx

```diff
@@ -0,0 +1,54 @@
+import { currentLanguage } from 'scrivito'
+import { pseudoRandom32CharHex } from '../utils/pseudoRandom32CharHex'
+import { toast } from 'react-toastify'
+
+export async function notifyOnConnectionTimeout<T>(connection: Promise<T>) {
+  const connectionSucceeded = await Promise.race([
+    connection.then(() => true),
+    waitFor(5).then(() => false),
+  ])
+  if (connectionSucceeded) return
+
+  const toastId = pseudoRandom32CharHex()
+
+  toast.warning(
+    <div>
+      <div>{localizeUnableToConnect()}</div>
+      <small>{localizePisaSalesServerRunning()}</small>
+    </div>,
+    { autoClose: false, toastId },
+  )
+
+  await connection
+  toast.dismiss(toastId)
```

Comment:
* [ ] FYI Maybe change the behavior to how `toast.promise` behaves by default: If the promise toast resolves, there is a success message. This will make the state change clearer than just hiding the toast. For example, the visitor may not have had enough time to read the message of the "pending" toast.